Assets/MRTK/Providers/LeapMotion/LeapMotionArticulatedHand.cs

```diff
@@ -0,0 +1,299 @@
+            if (Application.isPlaying)
+            {
+                // Get the attachment hand data based on controller handedness
+                attachmentHand = Array.Find(AttachmentHands.attachmentHands, (hand => HandChiralityToHandedness(hand.chirality) == ControllerHandedness));
+
+                // Enable all attachment point flags in the leap hand. By default, only the wrist and the palm are enabled.
+                foreach (TrackedHandJoint joint in Enum.GetValues(typeof(TrackedHandJoint)))
+                {
+                    AttachmentHands.attachmentPoints |= ConvertMRTKJointToLeapJoint(joint);
+                }            
+            }
+        }
+#endif
+        public override bool TryGetJoint(TrackedHandJoint joint, out MixedRealityPose pose)
+        {
+#if LEAPMOTIONCORE_PRESENT
+            if (attachmentHand != null && attachmentHand.isTracked)
+            {
+                // Is the current joint a metacarpal
+                bool isMetacarpal = Array.Exists(metacarpals, (metacarpalJoint => metacarpalJoint == joint));
+
+                // AttachmentPointFlags does not include metacarpals.  
+                if (isMetacarpal) 
+                {
+                    MixedRealityPose metacarpalPose = GetMetacarpals(joint);
+
+                    pose = metacarpalPose;
+                    return true;
+                }
+
+                AttachmentPointFlags leapAttachmentFlag = ConvertMRTKJointToLeapJoint(joint);
+
+                // Get the pose of the leap joint
+                AttachmentPointBehaviour leapJoint = attachmentHand.GetBehaviourForPoint(leapAttachmentFlag);
+
+                // Set the pose calculated by the leap motion to a mixed reality pose
+                pose = new MixedRealityPose(leapJoint.transform.position, leapJoint.transform.rotation);
+                return true;
+            }          
+#endif
+            pose = MixedRealityPose.ZeroIdentity;
+            return false;
+        }
+#if LEAPMOTIONCORE_PRESENT
+
+        // Convert leap hand chirality to MRTK handedness
+        private Handedness HandChiralityToHandedness(Chirality chirality)
+        {
+            return (chirality == Chirality.Left) ? Handedness.Left : Handedness.Right;
+        }
+
+        /// <summary>
+        /// Get the pose of the metacarpal joints from the current frame of the LeapServiceProvider because the metacarpal joints 
+        /// are not included in AttachmentPointFlags.  The metacarpal joints are those located directly above the wrist.
+        /// </summary>
+        /// <param name="metacarpalJoint"></param>
+        /// <returns>The MixedRealityPose for the leap metacarpal joint</returns>
+        private MixedRealityPose GetMetacarpals(TrackedHandJoint metacarpalJoint)
+        {
+            // Get the leap hand data in the current frame
+            List<Hand> hands = LeapMotionDeviceManager.CurrentHandsDetectedByLeap;
```

Comment:
Instead of asking the device manager for all hands and finding itself when needed, could the correct `Hand` be passed in the `UpdateState()` method and cached?

Or, does it need to be set multiple times? Could it be cached in the constructor and re-used?

Reply:
The leap `Hand` can be cached in the LeapMotionArticulatedHand constructor.  Suggestion has been included.